source/ci_source/providers/Codeship.ts

```diff
@@ -0,0 +1,40 @@
+import { Env, CISource } from "../ci_source"
+import { ensureEnvKeysExist, getPullRequestIDForBranch } from "../ci_source_helpers"
+
+export class Codeship implements CISource {
+  private default = { prid: "0" }
+  constructor(private readonly env: Env) {
+  }
+
+  async setup(): Promise<any> {
+    const prid = await getPullRequestIDForBranch(this, this.env, this.branchName)
```

Comment:
any chance this can be `prID` (makes it more consistent with the rest of the file)